src/Symfony/Component/Serializer/Normalizer/PropertyNormalizer.php

```diff
@@ -32,6 +36,38 @@
  */
 class PropertyNormalizer extends AbstractObjectNormalizer
 {
+    /**
+     * Flag to control whether public fields should be output when normalizing or omitted.
+     */
+    public const NORMALIZE_PUBLIC = 'normalize_public';
+
+    /**
+     * Flag to control whether protected fields should be output when normalizing or omitted.
+     */
+    public const NORMALIZE_PROTECTED = 'normalize_protected';
+
+    /**
+     * Flag to control whether private fields should be output when normalizing or omitted.
+     */
+    public const NORMALIZE_PRIVATE = 'normalize_private';
+
+    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyTypeExtractorInterface $propertyTypeExtractor = null, ClassDiscriminatorResolverInterface $classDiscriminatorResolver = null, callable $objectClassResolver = null, array $defaultContext = [])
+    {
+        parent::__construct($classMetadataFactory, $nameConverter, $propertyTypeExtractor, $classDiscriminatorResolver, $objectClassResolver, $defaultContext);
+
+        if (!\array_key_exists(self::NORMALIZE_PUBLIC, $this->defaultContext)) {
```

Comment:
As we don't have to deal with `null` values, I'd rather use `isset` for a slight performance improvement.